Lib/_pyrepl/commands.py

```diff
@@ -435,9 +435,16 @@ def do(self) -> None:
 class help(Command):
     def do(self) -> None:
         import _sitebuiltins
+        import signal
 
-        with self.reader.suspend():
-            self.reader.msg = _sitebuiltins._Helper()()  # type: ignore[assignment, call-arg]
+        if self.reader.help_mode:
+            self.reader.help_mode = False
+            signal.raise_signal(signal.SIGINT)
```

Comment:
Hummm...I am worried by this. Why are we raising SIGINT here? This looks very error prone

Reply:
I do think this is the correct approach. Relying on signals for this is super error prone in my opinion. Not only that but testing this that way is going to be a nightmare taking into account how signals behave in different machines. Also if the user has handles already installed you will be executing that code which may have side effects that the user didn't trigger.

This probably requires some heavier changes but I don't think we want to deal with the chaos of control flow by signals. 

WDYT @ambv

Reply:
The newest version of the code doesn't `raise_signal` anymore but raises a regular exception with KeyboardInterrupt. This is a bit weird, but we already do this ourselves in `commands.ctrl_c`. That command is mapped to `\x03` in `reader` (used by the asyncio REPL) and in the `ExceptHookHandler` (used for `threading.excepthook` so that exceptions from threads created in the REPL format okay).

While in both cases the command works around the fact that there's no good way to interrupt input from another thread, the way we actually interrupt the input is by raising KeyboardInterrupt by hand in the command.

All I'm saying is that there's precedent. I would want to test this with the asyncio REPL.